### PR TITLE
Oneway service request from the command line

### LINKED ISSUE #97: Call one-way requester / responder from command line
**Original report ([archived issue](https://osrf-migration.github.io/ignition-gh-pages/#!/ignitionrobotics/ign-transport/issues/97)) by Louise Poubel (Bitbucket: [chapulina](https://bitbucket.org/%7B5cfa2075-477b-4ded-bdb9-8d2479544ec4%7D/), GitHub: [chapulina](https://github.com/chapulina)).**

----------------------------------------
It currently isn't possible to call a service which has no request or one that has no response from the command line, because it requires that all fields are set.

### PR DESCRIPTION
**Pull request by caguero**
# 🦟 Bug fix

Partially fixes #97

## Summary

This patch makes it possible to request a one-way service request from the command line. Before, it was possible passing `--reptype gz.msgs.Empty` but always getting the message `Service call timed out` as an output (because the service callback wasn't answering anything, which was expected).

## How to test it?

You can compile the examples, and then:

1. Launch the one-way responser:
```
./responser_oneway
```

2. Request the service using the two-way request and verify that you don't get the timeout:
```
gz service -s /oneway --reqtype gz.msgs.StringMsg --reptype gz.msgs.Empty --req 'data: "Hello"' --timeout 1000
```

3. Use the new one-way request and verify that you don't get the timeout message:
```
gz service -s /oneway --reqtype gz.msgs.StringMsg --req 'data: "Hello"'
``` 

Note that both requests should reach the responser:
```
Request received: [Hello]
Request received: [Hello]
```

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Comment by azeey** [comment]
Would it be possible to detect that the response type is `Empty` instead of relying on the timeout? I like the idea of making some of the arguments optional (see https://github.com/gazebosim/gz-transport/issues/475, https://github.com/gazebosim/gz-transport/issues/474, https://github.com/gazebosim/gz-transport/issues/473), but there is a lot of existing documentation that uses the ` --reptype gz.msgs.Empty`  and `--timeout` arguments. It would be nice if those commands didn't timeout as well.

**Comment by caguero** [comment]
> 26d669b

Sounds good to me. Then, now both:

```
gz service -s /oneway --reqtype gz.msgs.StringMsg --reptype gz.msgs.Empty --req 'data: "Hello"' --timeout 10
```
and
```
gz service -s /oneway --reqtype gz.msgs.StringMsg --req 'data: "Hello"'
```

shoudn't generate timeouts.

**Comment by caguero** [comment]
@j-rivero , do you know why the same job (Ubuntu CI / Ubuntu Jammy CI) can pass and fail at the same time? It seems that `GZ_PATH` and `g_gzVersion` is undefined.

**Comment by azeey** [comment]
There have been some changes in https://github.com/gazebosim/gz-transport/pull/481 that affect this PR. I suggest merging from `gz-transport13`.

**Comment by caguero** [comment]
Good to go again!

File: src/cmd/gz.hh [hunk 1]
```diff
@@ -44,24 +44,26 @@ extern "C" void cmdServiceList();
...
-                                                       const char *_msgData);
+                            const char *_msgType,
+                            const char *_msgData);
 
 /// \brief External hook to execute 'gz service -r' from the command line.
 /// \param[in] _service Service name.
 /// \param[in] _reqType Message type used in the request.
 /// \param[in] _repType Message type used in the response.
 /// \param[in] _timeout The request will timeout after '_timeout' ms.
+///                     If -1 is used, the request will be one-way and _repType
```
Comment (azeey) [inline src/cmd/gz.hh:55]: Users might expect a timeout of -1 as "wait forever", which would be the opposite of this. I would say just checking if the message type is `Empty` is sufficient, but if you think we should have the timeout check as well, we should definitely update the help message of the CLI to mention this.

**Review by azeey — APPROVED** [review]
LGTM! Just one comment about the timeout.

File: src/cmd/gz.hh (hunk 1 above)
Comment (caguero) [reply src/cmd/gz.hh]: I think it's fine just checking the response type. I updated it in [687edcc](https://github.com/gazebosim/gz-transport/pull/477/commits/687edcc8fc79919920f8b824d952d93472f87a49).

**Pull request merged**